sfs/mono/source.py

```diff
@@ -705,6 +705,99 @@ def plane_averaged_intensity(omega, x0, n0, grid, c=None):
     return util.XyzComponents([i * n for n in n0])
 
 
+def pulsating_sphere(omega, center, radius, amplitude, grid, c=None):
+    """Sound pressure of a pulsating sphere.
+
+    Parameters
+    ---------
+    omega : float
+        Frequency of pulsating sphere
+    center : (3,) array_like
+        Center of sphere.
+    radius : float
+        Radius of sphere.
+    amplitude : float
+        Amplitude of displacement.
+    grid : triple of array_like
+        The grid that is used for the sound field calculations.
+        See `sfs.util.xyz_grid()`.
+    c : float, optional
+        Speed of sound.
+
+    Returns
+    -------
+    numpy.ndarray
+        Sound pressure at positions given by *grid*.
+
+    """
+    if c is None:
+        c = defs.c
+    k = util.wavenumber(omega, c)
+    center = util.asarray_1d(center)
+    grid = util.as_xyz_components(grid)
+
+    distance = np.linalg.norm(grid - center)
+    theta = np.arctan(1, k * distance)
+    impedance = defs.rho0 * c * np.cos(theta) * np.exp(1j * theta)
+    radial_velocity = 1j * omega * amplitude * radius / distance \
+        * np.exp(-1j * k * (distance - radius))
+    radial_velocity[distance <= radius] = np.nan
+    return impedance * radial_velocity
+
+
+def pulsating_sphere_velocity(omega, center, radius, amplitude, grid, c=None):
+    """Particle velocity of a pulsating sphere.
+
+    Parameters
+    ---------
+    omega : float
+        Frequency of pulsating sphere
+    center : (3,) array_like
+        Center of sphere.
+    radius : float
+        Radius of sphere.
+    amplitude : float
+        Amplitude of displacement.
+    grid : triple of array_like
+        The grid that is used for the sound field calculations.
+        See `sfs.util.xyz_grid()`.
+    c : float, optional
+        Speed of sound.
+
+    Returns
+    -------
+    `XyzComponents`
+        Particle velocity at positions given by *grid*.
+
+    Examples
+    --------
+
+    .. plot::
+        :context: close-figs
+
+        radius = 0.25
+        amplitude = 1 / (radius * omega * sfs.defs.rho0 * sfs.defs.c)
+        p = sfs.mono.source.pulsating_sphere(omega, x0, radius, amplitude, grid)
+        v = sfs.mono.source.pulsating_sphere_velocity(omega, x0, radius, amplitude, vgrid)
```

Comment:
This line has more than 79 characters (more specifically it has 82) when shown in the docs.